### PR TITLE
Feat: Add new data quality button

### PR DESCRIPTION
**Pull request by mariaozamiz**
### :pushpin: References

-   **Issue:** Closes: [Create contextual menu for New Data Quality button, that lets you select 'Module 1' or 'Module 2' (Datasets)](https://app.clickup.com/t/8693qa4cq)

### :memo: Implementation

Add component based on [Material context menu](https://v4.mui.com/components/menus/#context-menu) but including buttons inside, at set it into the filters area of the table component.

### :video_camera: Screenshots/Screen capture
Captura de pantalla 2024-02-26 a las 19 38 33
Captura de pantalla 2024-02-26 a las 19 38 39

### :fire: Notes to the tester
Check it out at dashboard page

### CODE REVIEW

File: src/webapp/components/menu-button/MenuButton.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            <Menu
+                id="simple-menu"
+                anchorEl={anchorEl}
+                keepMounted
+                open={Boolean(anchorEl)}
+                onClose={handleClose}
+            >
+                {items.map(item => (
+                    <MenuItem key={item.id} onClick={handleClose}>
+                        <Button variant="text" color="primary" onClick={item.handleClick}>
```
Comment (eperedo) [inline src/webapp/components/menu-button/MenuButton.tsx:54]: I think this component should expose what `Item` was selected so the parent component can decide what to do instead passing a `handleClick` for each item:

```tsx
type Props = { label: string; items: Item[]; onItemSelected: (id: string) => void };

const handleClick = (item: Item) => {
  onItemSelected(item.id);
};

<Button variant="text" color="primary" onClick={() => handleClick(item)}>
```

Then in the parent component:

```tsx
const menuButtonItems = [
    {
        label: "Module 1",
        id: "module-1",
        handleClick: () => alert("Module 1"),
    },
    {
        label: "Module 2",
        id: "module-2",
        handleClick: () => alert("Module 2"),
    },
];

const onMenuItemSelected= (id: Id) => {
    const selectedItem= menuButtonItems.find(item=> item.id === id);
    // do something with the item
};

<MenuButton
    label={i18n.t("New Data Quality")}
    items={menuButtonItems}
    onItemSelected={onMenuItemSelected}
/>
```

does it make sense?

File: src/webapp/pages/dashboard/hooks/useDashboard.tsx [hunk 2]
```diff
@@ -70,6 +84,7 @@ export function useDashboard() {
                 value={"somevalue"}
                 label={i18n.t("Status")}
             />
+            <MenuButton label={"New Data Quality"} items={menuButtonItems} />
```
Comment (eperedo) [inline src/webapp/pages/dashboard/hooks/useDashboard.tsx:87]: i18n for the string
```tsx
<MenuButton label={i18n.t("New Data Quality")} items={menuButtonItems} />
```

**Review by eperedo — CHANGES_REQUESTED** [review]
Thanks @mariaozamiz, code looks good just a minor refactor to avoid passing an extra method for each item in the `MenuButton` component. Let me know if you have any questions.

File: src/webapp/pages/dashboard/hooks/useDashboard.tsx (hunk 2 above)
Comment (mariaozamiz) [reply src/webapp/pages/dashboard/hooks/useDashboard.tsx:87]: In this case, translation was set in the original component, but should we translate in the instances?

//MenuButton.tsx

```
       <Button
                aria-controls="simple-menu"
                aria-haspopup="true"
                variant="contained"
                color="primary"
                onClick={onOpenMenu}
                endIcon={anchorEl ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
            >
                {i18n.t(label)}
            </Button>
```

Comment (eperedo) [reply src/webapp/pages/dashboard/hooks/useDashboard.tsx]: Yes, i18n needs to know in advance what's the string you want to translate (when running `yarn localize`) in order to generate them into *.po files. In this case we can remove i18n in the component.

**Review by Ramon-Jimenez — COMMENTED** [review]
Looks great @mariaozamiz

**Pull request merged**